src/unit/test_hashset.c

```diff
@@ -377,36 +377,36 @@ int test_iterator(int argc, char **argv, int flags) {
     UNUSED(argv);
     UNUSED(flags);
 
-    long count = 2000000;
+    size_t count = 2000000;
+    uint8_t element_array[count];
+    memset(element_array, 0, sizeof element_array);
 
-    /* A set of longs, i.e. pointer-sized values. */
+    /* A set of uint8_t pointers */
     hashsetType type = {0};
     hashset *s = hashsetCreate(&type);
-    long j;
 
     /* Populate */
-    for (j = 0; j < count; j++) {
-        assert(hashsetAdd(s, (void *)j));
+    for (size_t j = 0; j < count; j++) {
+        assert(hashsetAdd(s, element_array + j));
     }
 
     /* Iterate */
-    uint8_t element_returned[count];
-    memset(element_returned, 0, sizeof element_returned);
-    long num_returned = 0;
+    size_t num_returned = 0;
     hashsetIterator iter;
     hashsetInitIterator(&iter, s);
-    while (hashsetNext(&iter, (void **)&j)) {
+    uint8_t *item;
+    while (hashsetNext(&iter, (void **)&item)) {
         num_returned++;
-        assert(j >= 0 && j < count);
-        element_returned[j]++;
+        assert(item >= element_array && item < element_array + count);
+        (*item)++;
```

Comment:
Now, the array name `element_returned` is lost, so it can be good to explain in another way what we're doing here. A comment like this?

```suggestion
        /* Increment the element at this position, as a counter
         * of how many times this element has been returned. */
        (*item)++;
```

Another question: Why mix the words "item" and "element"? They mean the same. Let's use element?